client/my-sites/themes/index.node.js

```diff
@@ -12,21 +11,42 @@ import {
 	redirectFilterAndType,
 	redirectToThemeDetails
 } from './controller';
-import validateFilters from './validate-filters';
+import { validateFilters, validateVertical } from './validate-filters';
 
 export default function( router ) {
-	const verticals = getSubjects().join( '|' );
-
 	if ( config.isEnabled( 'manage/themes' ) ) {
 		// Redirect interim showcase route to permanent one
 		router( [ '/design', '/design/*' ], ( { originalUrl, res } ) => {
 			res.redirect( 301, '/themes' + originalUrl.slice( '/design'.length ) );
 		} );
 
-		router( `/themes/:vertical(${ verticals })?/:tier(free|premium)?`, fetchThemeFilters, fetchThemeData, loggedOut, makeLayout );
 		router(
-			`/themes/:vertical(${ verticals })?/:tier(free|premium)?/filter/:filter`,
+			'/themes/:tier(free|premium)?',
+			fetchThemeFilters,
+			fetchThemeData,
+			loggedOut,
+			makeLayout
+		);
+		router(
+			'/themes/:tier(free|premium)?/filter/:filter',
+			fetchThemeFilters,
+			validateFilters,
+			fetchThemeData,
+			loggedOut,
+			makeLayout
+		);
+		router(
+			'/themes/:vertical?/:tier(free|premium)?',
+			fetchThemeFilters,
+			validateVertical,
+			fetchThemeData,
+			loggedOut,
+			makeLayout
+		);
+		router(
+			'/themes/:vertical?/:tier(free|premium)?/filter/:filter',
 			fetchThemeFilters,
+			validateVertical,
 			validateFilters,
 			fetchThemeData,
 			loggedOut,
```

Comment:
We could collapse all route defs into one (by using an array of route def strings) if we modify the middlewares to work  with no verticals/filters respectively. Up to you tho.